src/Knp/Component/Pager/Event/Subscriber/Sortable/ArraySubscriber.php

```diff
@@ -2,82 +2,93 @@
 
 namespace Knp\Component\Pager\Event\Subscriber\Sortable;
 
-use \ReflectionClass;
-use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Knp\Component\Pager\Event\ItemsEvent;
-use Doctrine\ORM\Query;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
 
 class ArraySubscriber implements EventSubscriberInterface
 {
-
     /**
      * @var string the field used to sort current object array list
      */
-    private $currentSortingFieldGetter;
+    private $currentSortingField;
 
     /**
      * @var string the sorting direction
      */
     private $sortDirection;
 
-    public function items(ItemsEvent $event)
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    public function __construct(PropertyAccessorInterface $accessor = null)
     {
-        if (is_array($event->target) && count($event->target) > 1)
```

Comment:
Refactoring and enhancing in the same PR makes the diff roughly hard to read. Please next time create two separate PRs, it will make our life's easier.